tests/tools/test_hashline.py

```
@@ -0,0 +1,142 @@
+    def test_returns_two_hex_chars(self) -> None:
+        result = content_hash("hello")
+        assert len(result) == HASH_LENGTH
+        # Must be valid hex
+        int(result, 16)
+
+    def test_deterministic(self) -> None:
+        assert content_hash("test") == content_hash("test")
+
+    def test_matches_md5_prefix(self) -> None:
+        line = "function hello() {"
+        expected = hashlib.md5(line.encode()).hexdigest()[:HASH_LENGTH]
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)